recipes/gdcm/all/conanfile.py

```diff
@@ -25,101 +30,89 @@ class GDCMConan(ConanFile):
 
     def requirements(self):
-        self.requires("expat/2.4.8")
-        self.requires("openjpeg/2.4.0")
+        self.requires("expat/2.4.9")
+        self.requires("openjpeg/2.5.0")
         self.requires("zlib/1.2.12")
 
     def validate(self):
-        if self.options.shared and self._is_msvc and "MT" in msvc_runtime_flag(self):
-            raise ConanInvalidConfiguration("shared gdcm can't be built with MT or MTd")
-        if self.settings.compiler.get_safe("cppstd"):
-            tools.check_min_cppstd(self, "11")
+        if self.info.settings.compiler.cppstd:
+            check_min_cppstd(self, self._minimum_cpp_standard)
+        if is_msvc_static_runtime(self) and self.info.options.shared:
+            raise ConanInvalidConfiguration(f"{self.ref} does not support shared and static runtime together.")
 
     def source(self):
-        tools.get(**self.conan_data["sources"][self.version],
-                  destination=self._source_subfolder, strip_root=True)
-
-    def _patch_sources(self):
-        for patch in self.conan_data.get("patches", {}).get(self.version, []):
-            tools.patch(**patch)
-
-    def _configure_cmake(self):
-        if self._cmake:
-            return self._cmake
+        get(self, **self.conan_data["sources"][self.version],
+                  destination=self.source_folder, strip_root=True)
 
-        self._cmake = CMake(self)
-        self._cmake.definitions["GDCM_BUILD_DOCBOOK_MANPAGES"] = False
-        self._cmake.definitions["GDCM_BUILD_SHARED_LIBS"] = self.options.shared
+    def generate(self):
+        tc = CMakeToolchain(self)
+        tc.cache_variables["GDCM_BUILD_DOCBOOK_MANPAGES"] = False
+        tc.cache_variables["GDCM_BUILD_SHARED_LIBS"] = self.info.options.shared
         # FIXME: unvendor deps https://github.com/conan-io/conan-center-index/pull/5705#discussion_r647224146
-        self._cmake.definitions["GDCM_USE_SYSTEM_EXPAT"] = True
-        self._cmake.definitions["GDCM_USE_SYSTEM_OPENJPEG"] = True
-        self._cmake.definitions["GDCM_USE_SYSTEM_ZLIB"] = True
-        if not tools.valid_min_cppstd(self, 11):
-            self._cmake.definitions["CMAKE_CXX_STANDARD"] = 11
-
-        self._cmake.configure(build_folder=self._build_subfolder)
-        return self._cmake
+        tc.cache_variables["GDCM_USE_SYSTEM_EXPAT"] = True
+        tc.cache_variables["GDCM_USE_SYSTEM_OPENJPEG"] = True
+        tc.cache_variables["GDCM_USE_SYSTEM_ZLIB"] = True
+        if not self.settings.compiler.cppstd:
+            tc.blocks["cppstd"].values = { "cppstd": self._minimum_cpp_standard, "cppstd_extensions": "OFF" }
+        tc.generate()
+        deps = CMakeDeps(self)
+        previous_value = self.dependencies["expat"].cpp_info.get_property("cmake_find_mode")
+        self.dependencies["expat"].cpp_info.set_property("cmake_find_mode", "module")
+        deps.generate()
+        self.dependencies["expat"].cpp_info.set_property("cmake_find_mode", previous_value)
```

Comment:
`self.dependencies` is read-only: https://docs.conan.io/en/latest/reference/conanfile/dependencies.html#dependencies-interface

Reply:
fixed